pages/docs/tutorial/time_travel.mdx

```diff
@@ -16,7 +16,110 @@ To continue editing, reattach the document to the latest version using
 `doc.attach()`. This design is temporary and will be phased out once we have a
 more refined version control API in place.
 
-Below is an example demonstrating Time Travel functionality.
+## Read-only Time Travel
+
+Below we demonstrate how to implement simple, read-only time-travel. You could,
+for example, combine this with a slider in a UI to allow users to view the document
+over time.
+
+### Enable Timestamps
+
+Before this example will work, it is important that the edits made to the document
+have had [timestamp storage](/docs/advanced/timestamp) enabled:
+
+```ts
+doc.setRecordTimestamp(true);
+```
+
+This makes sure that all changes to the document will have a timestamp added to it.
+We will use this timestamp to sort changes so that the ordering will match user
+intuition.
+
+### Implementing Time Travel
+
+The first step is to load our document. Here we assume that you have a snapshot from your database
+or API.
+
+```ts
+// Get the snapshot for your doc from your database / API
+let snapshot = fetchSnapshot();
+
+// Import into a new document
+const doc = new LoroDoc();
+doc.import(snapshot);
+```
+
+Next we must collect and sort the timestamps for every change in the document. We want uesrs to be
+able to drag a slider to select a timestamp out of this list.
+
+```ts
+// Collect all changes from the document
+const changes = doc.getAllChanges();
+
+// Get the timestamps for all changes
+const timestamps = [...changes.values()]
```

Comment:
Maybe we can use a set to make the timestamps unique:

```
const timestamps = Array.from(new Set([...changes.values()]
  .flat() // Flatten changes from all peers into one list
  .map((x) => x.timestamp) // Get the timestamp from each peer
  .filter((x) => !!x))); 
```

Reply:
Updated!